tests/geometry/test_tools.py

```diff
@@ -519,3 +528,70 @@ def test_ValueError_given_inputs_axis_0_size_not_2(self, size, vector_name):
 
         with pytest.raises(ValueError):
             find_clockwise_angle_2d(**params)
+
+
+@debug_naughty_geometry
+def naughty_function(wire, var=1, *, var2=[1, 2], **kwargs):
+    raise cadapi.FreeCADError
+
+
+@debug_naughty_geometry
+def naughty_function_fallback(wire, var=1, *, var2=[1, 2], **kwargs):
+    try:
+        raise cadapi.FreeCADError
+    except cadapi.FreeCADError:
+        result = 42
+        raise _FallBackError(result=result)
+
+
+class TestDebugNaughtyGeometry:
+    path = get_bluemira_path("generated_data/naughty_geometry", subfolder="")
+
+    wires = [
+        make_polygon({"x": [0, 2, 2, 0], "y": [-1, -1, 1, 1]}, closed=True),
+        make_circle(),
+        make_ellipse(),
+        PrincetonD().create_shape(),
+    ]
+
+    @pytest.mark.parametrize("wire", wires)
+    def test_file_is_made(self, wire):
```

Comment:
I'm not a massive fan of using I/O in tests. It's slow and also opens the test suite up to race conditions when tests are run in parallel, which as the repo grows, we'll probably want to do if we aren't already. In fact, running this with 2 threads, the test class failed 5/5 times for me.

The usual solution is to mock out `builtins.open`. Is that doable here?

Reply:
```
with patch("builtins.open") as file:
    with pytest.raises(cadapi.FreeCADError):
```

then file is a mocked file writer so you can check what it is called with to re extract you data. Possibly with https://docs.python.org/3.8/library/unittest.mock.html#unittest.mock.Mock.call_args_list